integration/test_client.py

```diff
@@ -24,12 +27,37 @@ class TestClient(IntegrationTestCase):
     """Tests for `Client`."""
 
     def test_authenticate(self):
-        # This is another test with multiple assertions, as it is a test of
-        # flow, rather than a single source of functionality.
         client = pylxd.Client("https://127.0.0.1:8443/", verify=False)
 
         self.assertFalse(client.trusted)
 
         client.authenticate("password")
 
         self.assertTrue(client.trusted)
+
+    def test_authenticate_with_project(self):
+        # FIXME: Once the client supports creating projects we should use that
+        #        and remove this hack.
+        os.system("lxc project create test-project")
+        self.addCleanup(os.system, "lxc project delete test-project")
+
+        client = pylxd.Client(
+            "https://127.0.0.1:8443/", verify=False, project="test-project"
+        )
+
+        self.assertFalse(client.trusted)
```

Comment:
For some reason client.trusted in True here and in the following test. I can't figure out why, but I think there might be something wrong with this patch as it is False in the above test (which doesn't provide a project).

https://pastebin.canonical.com/p/7VT6QkSZ7M/